### PR TITLE
Add benchmarks using benchee

### PR DESCRIPTION
**Pull request by viniciusmuller**
This pull request uses the already existing `10000_sales.csv` file and adds some benchmarks using benchee for the ease of performance testing in development

```
$ mix run bench/benchmark.exs
Operating System: Linux
CPU Information: AMD Ryzen 7 5700X 8-Core Processor
Number of Available Cores: 16
Available memory: 31.26 GB
Elixir 1.14.4
Erlang 25.3.2

Benchmark suite executing with the following configuration:
warmup: 2 s
time: 5 s
memory time: 0 ns
reduction time: 0 ns
parallel: 1
inputs: none specified
Estimated total run time: 21 s

Benchmarking 10k lines, 14 columns - NimbleCSV.RFC4180.parse_enumerable/1 ...
Benchmarking 10k lines, 14 columns - NimbleCSV.RFC4180.parse_stream/1 ...
Benchmarking 10k lines, 14 columns - NimbleCSV.RFC4180.parse_string/1 ...

Name                                                                   ips        average  deviation         median         99th %
10k lines, 14 columns - NimbleCSV.RFC4180.parse_enumerable/1         87.71       11.40 ms    ±10.27%       11.03 ms       16.41 ms
10k lines, 14 columns - NimbleCSV.RFC4180.parse_string/1             70.96       14.09 ms    ±11.61%       13.65 ms       20.89 ms
10k lines, 14 columns - NimbleCSV.RFC4180.parse_stream/1             18.35       54.49 ms    ±11.58%       51.62 ms       66.04 ms

Comparison:
10k lines, 14 columns - NimbleCSV.RFC4180.parse_enumerable/1         87.71
10k lines, 14 columns - NimbleCSV.RFC4180.parse_string/1             70.96 - 1.24x slower +2.69 ms
10k lines, 14 columns - NimbleCSV.RFC4180.parse_stream/1             18.35 - 4.78x slower +43.09 ms

```

### CODE REVIEW

File: bench/benchmark.exs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+sales_csv = "bench/data/10000_sales.csv"
+sales_string = File.read!(sales_csv)
+sales_list = String.split(sales_string, "\n")
+sales_stream = File.stream!(sales_csv)
```
Comment (josevalim) [inline bench/benchmark.exs:4]: Maybe this should be something like:

```suggestion
sales_stream = Stream.map(sales_list, & &1)
```

So the cost of reading the file is not included in the benchmark?

Comment (viniciusmuller) [reply bench/benchmark.exs:4]: Makes sense, I tried to move reading the file to outside the benchmarks but forgot about the fact that the stream will read it on demand

File: bench/benchmark.exs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+sales_csv = "bench/data/10000_sales.csv"
+sales_string = File.read!(sales_csv)
+sales_list = String.split(sales_string, "\n")
+sales_stream = Stream.map(sales_list, & &1)
```
Comment (josevalim) [inline bench/benchmark.exs:4]:
```suggestion
sales_stream = sales_list |> Enum.map(& &1 <> "\n") |> Stream.map(& &1)
```

Maybe do a quick check to ensure they all return the same result? I think they currently don't.

Comment (viniciusmuller) [reply bench/benchmark.exs:4]: Currently this expression outputs `true`:

```ex
sales_stream = Stream.map(sales_list, & &1)

alias NimbleCSV.RFC4180, as: CSV
dbg(CSV.parse_enumerable(sales_list) == (CSV.parse_stream(sales_stream)) |> Enum.to_list())
```

So I don't think we need to add these newlines to the stream

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

File: bench/benchmark.exs (hunk 2 above)
Comment (josevalim) [reply bench/benchmark.exs:4]: Ah, we treat the newlines in the stream as optional, perfect.

**Pull request merged**